web/Services/AzureStorage.cs

```diff
@@ -0,0 +1,124 @@
+                await CreateContainerAsync(varType, id);
+                containerClient = new BlobContainerClient(_storageConnectionString, containerName);
+            }
+            return containerClient;
+        }
+
+        public async Task CreateContainerAsync(string varType, int id)
+        {
+            string containerName = GetContainerName(varType, id);
+            BlobServiceClient blobServiceClient = new BlobServiceClient(_storageConnectionString);
+            var container = blobServiceClient.GetBlobContainerClient(containerName);
+            if (container.Exists())
+            {
+                await DeleteContainerAsync(varType, id);
+            }
+            await blobServiceClient.CreateBlobContainerAsync(containerName);
+        }
+        
+        public async Task DeleteContainerAsync(string varType, int id)
+        {
+            var containerClient = await GetContainerClient(varType, id);
+            await containerClient.DeleteAsync();
+        }
+
+        public async Task UploadFileAsync(IFormFile blob, string varType, int id)
+        {
+            var containerClient = await GetContainerClient(varType, id);
+            var blobClient = containerClient.GetBlobClient(blob.FileName);
+            int i = 0;
+            while (blobClient.Exists())
+            {
+                i += 1;
+                string fileName = SplitFileName(blob.FileName).name + i + "." + SplitFileName(blob.FileName).type;
+                blobClient = containerClient.GetBlobClient(fileName);
+            }
+            await using (Stream? data = blob.OpenReadStream())
+            {
+                await blobClient.UploadAsync(data);
+            }
+        }
+
+        public async Task<List<string>> ListFilesAsync(string varType, int id)
+        {
+            var containerClient = await GetContainerClient(varType, id);
+            List<string> fileNames = new List<string>();
+            fileNames.Add(id.ToString());
+            await foreach (BlobItem file in containerClient.GetBlobsAsync())
+            {
+                fileNames.Add(file.Name);
+            }
+            return fileNames;
+        }
+
+        public async Task<(Stream, string)> DownloadFileAsync(string varType, int id, string blobName)
+        {
+            var containerClient = await GetContainerClient(varType, id);
+            BlobClient blobClient = containerClient.GetBlobClient(blobName);
+            if (!blobClient.Exists())
+            {
+                var newContainerClient = new BlobContainerClient(_storageConnectionString, "default");
+                blobName = "error.png";
+                blobClient = newContainerClient.GetBlobClient(blobName);
+            }
+            Stream data = await blobClient.OpenReadAsync();
+            string fileType = SplitFileName(blobName).type;
+            return (data, fileType);
+        }
+
+        public async Task DeleteFileAsync(string varType, int id, string blobName)

```

Comment:
Don't feel like you need to call everything "SomethingAsync" unless you're providing both async and non-asnyc methods (which we don't need to)